In [21]:
import pandas as pd
from sklearn import linear_model


df = pd.read_csv("C:\\Users\\tasweemb\\Documents\\Machine Learning Module\\Datasets\\Auto\\auto-mpg.csv")

# Use 'weight' and 'displacement' as predictors
x = df[['weight', 'displacement']]
# Use 'mpg' as the target variable
y = df['mpg']

# Create a linear regression model
regr = linear_model.LinearRegression()
regr.fit(x, y)

# Predict the mpg of a car with weight = 2300kg and displacement = 1300cm3
predicted_mpg = regr.predict([[3000, 250]])

predicted_mpg

C:\Users\tasweemb\AppData\Local\anaconda3\lib\site-packages\sklearn\base.py:420: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([22.41764297])

For a car with a displacement of 250.0 cubic inches and a weight of 3000 pounds, the predicted miles per gallon (mpg) is approximately 22.42 mpg.

In [19]:
print(regr.coef_)

[-0.00579774 -0.01635835]


In [20]:
predicted_mpg = regr.predict([[3500, 300]])

print(predicted_mpg)

[18.70085323]


C:\Users\tasweemb\AppData\Local\anaconda3\lib\site-packages\sklearn\base.py:420: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


For a car with a displacement of 300.0 cubic inches and a weight of 3500 pounds, the predicted miles per gallon (mpg) is approximately 18.70 mpg.